### PR TITLE
GCP Pub/Sub Spec

### PR DESCRIPTION
**Pull request by patnaikshekhar**
Added documentation in new format for the GCP Pub/Sub scaler.

### CODE REVIEW

**Comment by patnaikshekhar** [comment]
@tomkerkhove can you review this PR instead of the previous one.

File: spec/README.md [hunk 1]
```diff
@@ -11,4 +11,5 @@ Here is an overview:
     - [Azure Service Bus Queue/Topic](./triggers/azure-service-bus.md)
     - [Azure Storage Queue](./triggers/azure-storage-queue.md)
     - [RabbitMQ Queue](./triggers/rabbit-mq-queue.md)
-    - [Redis](./triggers/redis.md)
\ No newline at end of file
+    - [Redis](./triggers/redis.md)
+    - [GCP Pub/Sub](./triggers/gcp-pub-sub.md)
```
Comment (tomkerkhove) [inline spec/README.md]: Nitpicking but can you add it to the list in alphabetical order please?

File: spec/triggers/gcp-pub-sub.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#  Google Cloud Platform‎ Pub/Sub
+
+This specification describes the `gcp-pub-sub` trigger.
+
+```yaml
+triggers:
+- type: gcp-pubsub
+  metadata:
+    subscriptionSize: "5"
```
Comment (tomkerkhove) [inline spec/triggers/gcp-pub-sub.md]: Nitpicking but maybe specify that this is optional and mention what the default value is?

File: spec/README.md (hunk 1 above)
Comment (patnaikshekhar) [reply spec/README.md]: @tomkerkhove this is fixed now.

File: spec/triggers/gcp-pub-sub.md (hunk 2 above)
Comment (patnaikshekhar) [reply spec/triggers/gcp-pub-sub.md]: @tomkerkhove this is fixed now.

File: spec/README.md (hunk 1 above)
Comment (tomkerkhove) [reply spec/README.md]: Thanks!

File: spec/triggers/gcp-pub-sub.md (hunk 2 above)
Comment (tomkerkhove) [reply spec/triggers/gcp-pub-sub.md]: Thanks!

**Comment by tomkerkhove** [comment]
:shipit:

**Pull request merged**